packages/headless/src/controllers/core/search-box/headless-core-search-box.ts

```diff
@@ -0,0 +1,271 @@
+import {
+  defaultSearchBoxOptions,
+  SearchBoxOptions,
+  searchBoxOptionsSchema,
+} from './headless-core-search-box-options';
+import {
+  SuggestionHighlightingOptions,
+  Delimiters,
+  getHighlightedSuggestion,
+} from '../../../utils/highlight';
+import {
+  buildController,
+  Controller,
+} from '../../controller/headless-controller';
+import {CoreEngine} from '../../..';
+import {
+  ConfigurationSection,
+  QuerySection,
+  QuerySetSection,
+  QuerySuggestionSection,
+  SearchSection,
+} from '../../../state/state-sections';
+import {
+  configuration,
+  query,
+  querySet,
+  querySuggest,
+  search,
+} from '../../../app/reducers';
+import {loadReducerError} from '../../../utils/errors';
+import {randomID} from '../../../utils/utils';
+import {validateOptions} from '../../../utils/validate-payload';
+import {
+  registerQuerySetQuery,
+  updateQuerySetQuery,
+} from '../../../features/query-set/query-set-actions';
+import {
+  clearQuerySuggest,
+  FetchQuerySuggestionsActionCreatorPayload,
+  registerQuerySuggest,
+  selectQuerySuggestion,
+} from '../../../features/query-suggest/query-suggest-actions';
+import {SearchAction} from '../../../features/analytics/analytics-utils';
+import {prepareForSearchWithQuery} from '../../../features/search/search-actions';
+import {logQuerySuggestionClick} from '../../../features/query-suggest/query-suggest-analytics-actions';
+import {logSearchboxSubmit} from '../../../features/query/query-analytics-actions';
+import {QuerySuggestState} from '../../../features/query-suggest/query-suggest-state';
+import {AsyncThunkAction} from '@reduxjs/toolkit';
+
+export type {SearchBoxOptions, SuggestionHighlightingOptions, Delimiters};
+
+export interface CoreSearchBoxProps {
+  /**
+   * The `CoreSearchBox` controller options.
+   */
+  options?: SearchBoxOptions;
+
+  executeSearchActionCreator?: (
```

Comment:
Why is this optional ? I think it should be required, and we should remove the check on line 182:

```typescript
if (props.executeSearchActionCreator) {
  await dispatch(props.executeSearchActionCreator(analytics));
}
```

Reasoning is that is should not be possible for a search box extending the core search box to be missing important functionality.

And if there is a good reason for a search box implementation to not expose a functioning `submit()` function that perform a search and retrieve new results, then it means the `submit()` function is probably not "core" for the search box, and that it should probably be removed from the `core-search-box`, and moved to each specific search box implementation.

Reply:
You are right, there is no good reason to make this action optional.